Library/src/main/java/com/shopify/testify/core/ScreenshotConfigurationInterface.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.shopify.testify.core
+
+import android.content.Context
+import android.graphics.Rect
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import com.shopify.testify.internal.modification.FocusModification
+import com.shopify.testify.internal.modification.HideCursorViewModification
+import com.shopify.testify.internal.modification.HidePasswordViewModification
+import com.shopify.testify.internal.modification.HideScrollbarsViewModification
+import com.shopify.testify.internal.modification.HideTextSuggestionsViewModification
+import com.shopify.testify.internal.modification.SoftwareRenderViewModification
+import com.shopify.testify.report.Reporter
+import java.util.Locale
+
+typealias ViewModification = (rootView: ViewGroup) -> Unit
```

Comment:
I know for the scope of this PR this is just being moved, but I'm curious as to why we have this `ViewModification` typealias when we also have the `com.shopify.testify.internal.modification.ViewModification` class, and they appear to do very similar things (although I guess some are more "first class" in terms of having autocomplete/api support). I guess my other question is why we only allow one to be set at a time.

Reply:
Yeah, this is an ugly holdover from the java & junit3 days. Basically, we had an internal set of functions that were all named "setBlahViewModification" and then also the public `setViewModification` method that's used in the test code. They were initially intended to be the same thing, but they kind of evolved a bit differently when we started refactoring to kotlin. In retrospect, we should never have used the same name for both things, or we should have made them literally the same. It's kind of techdebt, I guess.